code/modules/unit_tests/machine_disassembly.dm

```diff
@@ -0,0 +1,13 @@
+/// Ensures that when disassembling a machine, all the parts are given back
+/datum/unit_test/machine_disassembly/Run()
+	var/obj/machinery/freezer = allocate(/obj/machinery/atmospherics/components/unary/thermomachine/freezer)
+
+	var/turf/freezer_location = freezer.loc
+	freezer_location.ChangeTurf(/turf/open/floor/plasteel)
```

Comment:
If you're wondering, when you don't do this it only produces the frame. I have no idea why, and I haven't checked if it actually happens in game, but...yeah.

Reply:
spooky, but you could have added a comment and saved yourself this one :laughing: